lib/jxl/encode.cc

```diff
@@ -181,6 +181,19 @@ int VerifyLevelSettings(const JxlEncoder* enc, std::string* debug_string) {
   // All level 5 checks passes, so can return the more compatible level 5
   return 5;
 }
+JxlEncoderStatus CheckValidBitdepth(uint32_t bits_per_sample,
+                                    uint32_t exponent_bits_per_sample) {
+  if (!exponent_bits_per_sample) {
+    if (!(bits_per_sample > 0 && bits_per_sample <= 24)) {
```

Comment:
We could in principle allow up to 31 for the uint bitdepth, but our code cannot handle that atm (since everything goes through float at some point, which doesn't have enough mantissa bits for that). OK to keep it like this (though technically 25 to 31 wouldn't be invalid but just not supported).

Reply:
ok, added a comment